src/plugins/workspace/public/components/workspace_form/workspace_use_case.tsx

```diff
@@ -54,21 +57,49 @@ const WorkspaceUseCaseCard = ({
   );
 };
 
+type AvailableUseCase = Pick<WorkspaceUseCaseObject, 'id' | 'title' | 'description' | 'systematic'>;
+
 export interface WorkspaceUseCaseProps {
   value: string | undefined;
   onChange: (newValue: string) => void;
   formErrors: WorkspaceFormErrors;
-  availableUseCases: Array<
-    Pick<WorkspaceUseCaseObject, 'id' | 'title' | 'description' | 'systematic'>
-  >;
+  availableUseCases: AvailableUseCase[];
+  savedObjects: SavedObjectsStart;
+  operationType: WorkspaceOperationType;
 }
 
 export const WorkspaceUseCase = ({
   value,
   onChange,
   formErrors,
   availableUseCases,
+  savedObjects,
+  operationType,
 }: WorkspaceUseCaseProps) => {
+  const [displayedUseCases, setDisplayedUseCases] = useState(availableUseCases);
+  const [isOnlyAllowEssential, setIsOnlyAllowEssential] = useState(false);
+
+  useEffect(() => {
+    if (operationType === WorkspaceOperationType.Create) {
+      getIsOnlyAllowEssentialUseCase(savedObjects.client).then((result: boolean) => {
+        setIsOnlyAllowEssential(result);
+      });
+    }
+  }, [savedObjects, operationType]);
```

Comment:
It looks we can simply use `useMemo` instead of `useEffect` + `useState`

Reply:
Here `useMemo` would be hard to combine with promise.

Reply:
So I used `useEffect` + `useState` to make it clear.

Reply:
right, that's a promise